autodistill/core/embedding_ontology.py

```diff
@@ -13,6 +13,19 @@
     "EmbeddingOntologyText",
 ]
 
+def cosine_similarity(a: np.array, b: np.array) -> np.array:
+    """
+    Calculate the cosine similarity between two vectors.
+
+    Args:
+        a: The first vector.
+        b: The second vector.
+
+    Returns:
+        The cosine similarity between the two vectors.
+    """
+    return float(np.dot(a, b) / (np.linalg.norm(a) * np.linalg.norm(b)))
```

Comment:
Is it possible to do all the computations in torch 

    a = torch.from_numpy(a)
    b = torch.from_numpy(b)

    return float(torch.dot(a, b) / (torch.norm(a) * torch.norm(b))) somthing like as you don't have to move the embeddings to np and also GPU computation might be efficient

Reply:
can you sign the cla?

Reply:
just one more point, if the generated vectors return a zero norm in edge cases the cosine similarity will return Inf can you add
epsilon = 1e-10
cosine_similarity = float(np.dot(a, b) / (np.linalg.norm(a) * np.linalg.norm(b) + epsilon))

Reply:
Looks good now